src/ucp/proto/proto_am.inl

```diff
@@ -521,23 +509,34 @@ ucp_proto_get_zcopy_threshold(const ucp_request_t *req,
     return max_zcopy;
 }
 
+static UCS_F_ALWAYS_INLINE ssize_t
+ucp_proto_short_max_adjust(ssize_t max)
+{
+    /* short_iov which is used for compatibility path does not have header, so
+     * payload can be larger */
+    return max - sizeof(uint64_t);
+}
+
 static UCS_F_ALWAYS_INLINE ssize_t
 ucp_proto_get_short_max(const ucp_request_t *req,
                         const ucp_ep_msg_config_t *msg_config)
 {
     return (!UCP_DT_IS_CONTIG(req->send.datatype) ||
             (req->flags & UCP_REQUEST_FLAG_SYNC) ||
             (!UCP_MEM_IS_HOST(req->send.mem_type))) ?
-           -1 : msg_config->max_short;
+           -1 : ucp_proto_short_max_adjust(msg_config->max_short);
 }
 
 static UCS_F_ALWAYS_INLINE int
 ucp_proto_is_inline(ucp_ep_h ep, const ucp_memtype_thresh_t *max_eager_short,
                     ssize_t length)
 {
-    return (ucs_likely(length <= max_eager_short->memtype_off) ||
-            (length <= max_eager_short->memtype_on &&
-             ucp_memory_type_cache_is_empty(ep->worker->context)));
+    ucp_context_h context = ep->worker->context;
+
+    return (ucs_likely(length <=
+                       ucp_proto_short_max_adjust(max_eager_short->memtype_off)) ||
+            (length <= ucp_proto_short_max_adjust(max_eager_short->memtype_on) &&
```

Comment:
why needed if it's already done in ucp_ep_config_init?

Reply:
no need, lefover..